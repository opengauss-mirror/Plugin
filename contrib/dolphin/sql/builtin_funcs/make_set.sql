create schema make_set;
set current_schema to 'make_set';
set dolphin.sql_mode = '';
select make_set(3, 'a', 'b', 'c');
select make_set(2,'a','b','c','d','e');
select make_set(1|4, 'hello', 'nice', 'world');
select make_set(1|4, 'hello', 'nice', NULL, 'world');
select make_set(0, 'a', 'b', 'c');
select make_set('', 'a', 'b', 'c');
select make_set(null, 'a', 'b', 'c');

select make_set(2147483649, 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y');

select make_set(-1,'a','b','c','d','e');
select make_set(-2,'a','b','c','d','e');
select make_set(-3,'a','b','c','d','e');
select make_set(-4,'a','b','c','d','e');
select make_set(-5,'a','b','c','d','e');
select make_set(-6,'a','b','c','d','e');
select make_set(-7,'a','b','c','d','e');

select make_set(3,-1,4);

select make_set(7,-1,&#!#!@,'asfjoajfowqeknf,zncfiwuhrqnsdfhkasf');

select make_set(3,-1,&#!#!@);

select make_set(3,-1,'afasfasdfdsafsafasfdasdfsadfowijfqwjfowqfhasdhfsaqufhaksdfhskadfhasiufhakjfdahfiuwhfasjlajfowifjawefwfa');

select make_set(4294967296*1024*1024*1024,'1','2','3','4','5','6','7','8','9','10','11','12','13','14','15','16','17','18','19','20','21','22','23','24','25','26','27','28','29','30','31','32','33','34','35','36','37','38','39','40','41','42','43','44','45','46','47','48','49','50','51','52','53','54','55','56','57','58','59','60','61','62','63','64','65','66');

select make_set(-4294967296*1024*1024*1024,'1','2','3','4','5','6','7','8','9','10','11','12','13','14','15','16','17','18','19','20','21','22','23','24','25','26','27','28','29','30','31','32','33','34','35','36','37','38','39','40','41','42','43','44','45','46','47','48','49','50','51','52','53','54','55','56','57','58','59','60','61','62','63','64','65','66');

select make_set(3, true, false);
select make_set(3,01/02/03, false, true, false);

drop schema make_set cascade;
reset current_schema;

