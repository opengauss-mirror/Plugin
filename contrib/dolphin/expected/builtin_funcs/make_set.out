create schema make_set;
set current_schema to 'make_set';
set dolphin.sql_mode = '';
select make_set(3, 'a', 'b', 'c');
 make_set 
----------
 a,b
(1 row)

select make_set(2,'a','b','c','d','e');
 make_set 
----------
 b
(1 row)

select make_set(1|4, 'hello', 'nice', 'world');
  make_set   
-------------
 hello,world
(1 row)

select make_set(1|4, 'hello', 'nice', NULL, 'world');
 make_set 
----------
 hello
(1 row)

select make_set(0, 'a', 'b', 'c');
 make_set 
----------
 
(1 row)

select make_set('', 'a', 'b', 'c');
WARNING:  invalid input syntax for type bigint: ""
LINE 1: select make_set('', 'a', 'b', 'c');
                        ^
CONTEXT:  referenced column: make_set
 make_set 
----------
 
(1 row)

select make_set(null, 'a', 'b', 'c');
 make_set 
----------
 
(1 row)

select make_set(2147483649, 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y');
 make_set 
----------
 a
(1 row)

select make_set(-1,'a','b','c','d','e');
 make_set  
-----------
 a,b,c,d,e
(1 row)

select make_set(-2,'a','b','c','d','e');
 make_set 
----------
 b,c,d,e
(1 row)

select make_set(-3,'a','b','c','d','e');
 make_set 
----------
 a,c,d,e
(1 row)

select make_set(-4,'a','b','c','d','e');
 make_set 
----------
 c,d,e
(1 row)

select make_set(-5,'a','b','c','d','e');
 make_set 
----------
 a,b,d,e
(1 row)

select make_set(-6,'a','b','c','d','e');
 make_set 
----------
 b,d,e
(1 row)

select make_set(-7,'a','b','c','d','e');
 make_set 
----------
 a,d,e
(1 row)

select make_set(3,-1,4);
 make_set 
----------
 -1,4
(1 row)

select make_set(7,-1,&#!#!@,'asfjoajfowqeknf,zncfiwuhrqnsdfhkasf');
ERROR:  syntax error at or near ","
LINE 1: select make_set(7,-1,&#!#!@,'asfjoajfowqeknf,zncfiwuhrqnsdfh...
                                   ^
select make_set(3,-1,&#!#!@);
ERROR:  syntax error at or near ")"
LINE 1: select make_set(3,-1,&#!#!@);
                                   ^
select make_set(3,-1,'afasfasdfdsafsafasfdasdfsadfowijfqwjfowqfhasdhfsaqufhaksdfhskadfhasiufhakjfdahfiuwhfasjlajfowifjawefwfa');
                                                  make_set                                                  
------------------------------------------------------------------------------------------------------------
 -1,afasfasdfdsafsafasfdasdfsadfowijfqwjfowqfhasdhfsaqufhaksdfhskadfhasiufhakjfdahfiuwhfasjlajfowifjawefwfa
(1 row)

select make_set(4294967296*1024*1024*1024,'1','2','3','4','5','6','7','8','9','10','11','12','13','14','15','16','17','18','19','20','21','22','23','24','25','26','27','28','29','30','31','32','33','34','35','36','37','38','39','40','41','42','43','44','45','46','47','48','49','50','51','52','53','54','55','56','57','58','59','60','61','62','63','64','65','66');
 make_set 
----------
 63
(1 row)

select make_set(-4294967296*1024*1024*1024,'1','2','3','4','5','6','7','8','9','10','11','12','13','14','15','16','17','18','19','20','21','22','23','24','25','26','27','28','29','30','31','32','33','34','35','36','37','38','39','40','41','42','43','44','45','46','47','48','49','50','51','52','53','54','55','56','57','58','59','60','61','62','63','64','65','66');
 make_set 
----------
 63,64
(1 row)

select make_set(3, true, false);
 make_set 
----------
 1,0
(1 row)

select make_set(3,01/02/03, false, true, false);
      make_set       
---------------------
 0.166666666666667,0
(1 row)

drop schema make_set cascade;
reset current_schema;
